### ISSUE TITLE
Add Visual Studio Build Tools with Windows containers

### ISSUE THREAD

**Issue by hugobritobh** [clemlesne/azure-pipelines-agent#21]
If possible, it would be interesting in the Windows image to have BuildTools:

https://learn.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2022

**Comment by clemlesne** [comment]
[What component](https://learn.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-build-tools?view=vs-2022) would you require?

**Comment by hugobritobh** [comment]
For legacy project (Net Framework), Web Forms with Test project. Without Build Tools, no project was compiled.

I installed via choco the Build Tools with some components (It takes about 10 minutes):

```
$Components = @(
"Microsoft.Component.MSBuild",
"Microsoft.Net.Component.4.5.1.TargetingPack",
"Microsoft.Net.ComponentGroup.TargetingPacks.Common",
"Microsoft.Net.Component.4.8.SDK",
"Microsoft.Net.ComponentGroup.4.8.DeveloperTools",
"Microsoft.VisualStudio.Workload.ManagedDesktopBuildTools",
"Microsoft.VisualStudio.Workload.NodeBuildTools;includeOptional",
"Microsoft.VisualStudio.Web.BuildTools.ComponentGroup",
"Microsoft.VisualStudio.Workload.MSBuildTools",
"Microsoft.VisualStudio.Workload.WebBuildTools",
"Microsoft.VisualStudio.Component.TestTools.BuildTools",
"Microsoft.VisualStudio.Component.NuGet.BuildTools",
"Microsoft.VisualStudio.Component.WebDeploy",
"Microsoft.VisualStudio.Component.Roslyn.Compiler",
"Microsoft.VisualStudio.Component.Roslyn.LanguageServices",
"Microsoft.VisualStudio.Workload.DataBuildTools",
"Microsoft.VisualStudio.Component.SQL.SSDTBuildSku"
)

$InstallerArgs = "--add " + ($Components -join " --add ")

C:\ProgramData\chocolatey\bin\choco install visualstudio2022buildtools -y --params $InstallerArgs
```

Almost all projects now compile, only one test project didn't compile, but I think the problem may be there.

**Comment by clemlesne** [comment]
Branch `feat/win-build-tools` initiated. Can you test with the tag `ghcr.io/clemlesne/azure-pipelines-agent:win-ltsc2022-5.2.2-22.2a95a9d`?

**Comment by hugobritobh** [comment]
Normal   Pulling                 Pulling image "ghcr.io/clemlesne/azure-pipelines-agent:win-ltsc2022-5.2.2-22.2a95a9d"                                                                                                                                                
  Warning  Failed                 Error: ErrImagePull                                                          
 Normal   BackOff                Error: ImagePullBackOff

**Comment by clemlesne** [comment]
It seems GitHub container repo had a temporary failure. Can you retry with branch [feat/win-build-tools](https://github.com/clemlesne/azure-pipelines-agent/tree/feat/win-build-tools)?

Alternatively, [you can use Docker Hub](https://hub.docker.com/repository/docker/clemlesne/azure-pipelines-agent).

**Comment by clemlesne** [comment]
Branch merged: 02b686539d00d4cf92b43bef95f179d4784b1bed.
